### PR TITLE
Issue 11682: library_browser.js now handles the 'unwind' exception 

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: tests/emscripten_main_loop.cpp [hunk 1]
```diff
@@ -66,8 +66,12 @@ void looper() {
     assert(ret == 0);
   } else if (frame == 90) {
     emscripten_cancel_main_loop();
-    emscripten_set_main_loop(looper, 100, 1);
+    emscripten_set_main_loop(looper, 110, 1);
   } else if (frame == 100) {
+    emscripten_pause_main_loop();
+    emscripten_cancel_main_loop();
+    emscripten_set_main_loop(looper, 110, 1);
```
Comment (JCash) [inline tests/emscripten_main_loop.cpp:73]: Although there are are several "main loop" tests, I feel they test slightly different things, and also they use the "usual" style of coding.

I believe testing this issue in only this particular .cpp file suffices in this case.

**Comment by JCash** [comment]
Note that the test currently triggers the issue, but the test won't actually fail, because it doesn't fail on uncaught exceptions.
I'm afraid I currently don't know enough emscripten/test functionality to make the test _actually_ fail.
And to be honest, I would have thought it should have failed on it's own, since it's an uncaught exception?

Suggestions are welcome!

```
work/external/emscripten $ python tests/runner.py browser.test_emscripten_main_loop                                         [master] 12:33:14
hello, world!

tests/runner.py:WARNING: use EMTEST_ALL_ENGINES=1 in the env to run against all JS engines, which is slower but provides more coverage
Test suites:
['test_browser']
Running test_browser: (1 tests)
(checking sanity from test runner)
shared:INFO: (Emscripten: Running sanity checks)
[Browser harness server on process 51226]
/Users/mawe/work/external/emscripten/tests/runner.pyc:INFO: Using default system browser

Running the browser tests. Make sure the browser allows popups from localhost.

test_emscripten_main_loop (test_browser.browser) ... [client logging: /?exception=uncaught exception: unwind / undefined ]
[client logging: /?exception=uncaught exception: unwind / undefined ]
[client logging: /?exception=ReferenceError: SharedArrayBuffer is not defined / undefined ]
skipped 'SharedArrayBuffer is not supported!'
[Browser harness server terminated]

----------------------------------------------------------------------
Ran 1 test in 8.012s

OK (skipped=1)
```

**Comment by JCash** [comment]
Fixes https://github.com/emscripten-core/emscripten/issues/11682

**Comment by JCash** [comment]
Ping!

**Comment by kripken** [comment]
> [client logging: /?exception=ReferenceError: SharedArrayBuffer is not defined / undefined ]

I think that means the test is skipped as the browser doesn't support pthreads.

I verified that adding say `EM_ASM({ throw "error" })` makes the test fail when pthreads is present, so it should not be passing incorrectly, aside from ignoring non-pthreads browsers.

Was the testing issue just that, or something more? Not sure I understood what you are seeing there.

**Comment by JCash** [comment]
> Was the testing issue just that, or something more? Not sure I understood what you are seeing there.

It was just that, that my test didn't produce an error in my terminal.

**Comment by kripken** [comment]
@JCash Ok, sounds good, then I think that explain the issue then.

Will land as soon as tests pass. Thanks!

**Pull request merged**